pyanaconda/ui/tui/spokes/__init__.py

```diff
@@ -166,11 +166,12 @@ def prompt(self, entry = None):
             return _("Enter new value for '%s' and press enter\n") % entry.title
 
     def input(self, entry, key):
-        if entry.aux.match(key):
+        if entry.aux.match(key) and check_name(key)[0]:
             self.value = key
             self.close()
             return True
         else:
+            self.wrong_input_message = check_name(key)[1]
```

Comment:
No need to call ``check_name()`` twice here. Just store the result first and then use its contents like this:
```
(valid, err_msg) = check_name(key)
if entry.aux.match(key) and valid:
    ...
```